### DISCUSSION TITLE
[riverpod_gen] Here is a situation that I want to use a build return type which is provided by generated file.

### DISCUSSION THREAD

**Discussion by dddrop** [rrousselGit/riverpod#2406] [Q&A] [answered]
I want to create a provider which returns provider which generate by riverpod_gen:

```dart
@freezed
class Item with _$Item {
  const factory Item({
    required int id,
    required String name,
  }) = _Item;
}

@riverpod
class ItemManager extends _$ItemManager {
  @override
  Item build(Item item) => item;
}
```
Which creates a `ItemMangagerProvider`,  and

```dart
@riverpod
class GetItemUseCase extends _$GetItemUseCase {
  @override
  ItemManagerProvider build(int id) {
    return items[id]!;
  }
}
```

which should created `class GetItemUseCaseProvider extends AutoDisposeNotifierProviderImpl<GetItemUseCase, ItemManagerProvider> `, but actually it's generated as `class GetItemUseCaseProvider extends AutoDisposeNotifierProviderImpl<GetItemUseCase, dynamic>` which was a dynamic type.

**Comment by rrousselGit** [comment, marked as answer]
You shouldn't do that anyway.  
Don't abstract providers. Return the value returned by that provider instead

**Reply by dddrop** [reply]
Thanks.
I know do like that is bad way. And I also mean that, If a value class is generated in *.g.dart file. The situation above will happen.
